backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/ELKLayoutedDiagramProvider.java

```diff
@@ -76,11 +76,17 @@ private Node getLayoutedNode(Node node, ElkConnectableShape elkConnectableShape,
         Size size = Size.of(elkConnectableShape.getWidth(), elkConnectableShape.getHeight());
         Position position = Position.at(elkConnectableShape.getX(), elkConnectableShape.getY());
 
-        // @formatter:off
-        Label label = this.getLayoutedLabel(node.getLabel(), id2ElkGraphElements, 0, 0);
+        double xOffSet = 0;
+        if (!node.isBorderNode()) {
+            // The label is positioned at the center of the node and the front-end will apply a "'text-anchor':
+            // 'middle'" property.
+            xOffSet = node.getLabel().getSize().getWidth() / 2;
+        }
```

Comment:
Can make this hypothesis if in the class below, we have the code to deal with various label placement strategies? What was the issue with the previous algorithm?

Reply:
I did not catch your first sentence. 

> What was the issue with the previous algorithm?

The frontend needs to display the label area centered on the node and each line part centered inside the node
I used "text-anchor:middle" to do that. But to make it work the label position needs to be positionned on the center of the node. I tried to apply `text-anchor:middle` only on tspan instead of the parent text but `text-anchor:middle` make still the line positionned on the label x position